woocommerce/wc-admin
# Have NavigationNudge and NavigationFeedback self-delete when irrelevant

If the navigation feature does not exist, have the NavigationNudge note self-action.

Fixes #7807

_Replace this with a good description of your changes & reasoning._

### Screenshots

### Detailed test instructions:

- Start with a clean WooCommerce site
- Install WooCommerce Admin from this branch and activate it
- Verify that the Navigation checkbox appears on the Settings / Advanced / Features page.

#### NavigationNudge note

- If the 'You now have access to the WooCommerce navigation' note doesn't show up immediately, run the `wc_admin_daily` cron event. I'm using the WP Crontrol plugin.
- Remove the navigation feature (see below)
- Verify that the NavigationNudge note no longer appears (bonus points: check the database and verify that there is no row in `wp_wc_admin_notes` with `name = 'wc-admin-navigation-nudge'`).
- Reenable (but don't activate) the navigation feature: Just undo whatever you did two steps ago.
- Run the `wc_admin_daily` cron event
- Verify that the NavigationNudge note has been recreated

#### NavigationFeedback note

The NavigationFeedback note should be created by the `wc_admin_daily` cron event when the navigation feature is activated (the checkbox is checked on the Settings / Advanced / Features page), so go through the same steps as the NavigationNudge note but activate the navigation feature.

#### One way to remove the navigation feature:

Use the `'woocommerce_admin_features'` filter to remove the `'navigation'` filter like so (you can put this code anywhere it'll be executed--I'm using a custom plugin):

```
add_filter( 'woocommerce_admin_features', function ( $features ) {
	if (!empty($features)) {
		$features = array_filter($features, function ($feature) { return $feature !== 'navigation'; });
	}
	return $features;
} );
```
